framework/lib/components/switch/switch-field.tsx

```diff
@@ -19,25 +32,43 @@ export const SwitchField = ({
   validate,
   onChange: onChangeCallback = identity,
   direction = 'row',
+  labelPosition = 'right',
   ...rest
 }: SwitchFieldProps) => (
   <Box w={ label ? 'full' : 'fit-content' }>
     <Field
       name={ name }
-      label={ label }
+      label=""
       isRequired={ isRequired }
       direction={ direction }
       validate={ validate }
     >
-      { ({ onChange, value }) => (
-        <Switch
-          name={ name }
-          onChange={ (e) => { onChange(e); onChangeCallback(e) } }
-          value={ value }
-          data-testid="switch-field-test-id"
-          { ...rest }
-        />
-      ) }
+      { ({ onChange, value }) => {
+        const switchComponent = (
+          <Switch
+            name={ name }
+            onChange={ (e) => {
+              onChange(e)
+              onChangeCallback(e)
+            } }
+            value={ value }
+            data-testid="switch-field-test-id"
+            { ...rest }
+          />
+        )
+
+        const labelComponent = label ? <Label htmlFor={ name }>{ label }</Label> : null
+
+        return (
+          <Flex
+            gap={ 2 }
+            direction={ labelPosition === 'left' ? 'row-reverse' : 'row' }
+          >
+            { switchComponent }
```

Comment:
I would just do this :arrow_down: Assigning components to variables is harder to read
`````
          <Switch
            name={ name }
            onChange={ (e) => {
              onChange(e)
              onChangeCallback(e)
            } }
            value={ value }
            data-testid="switch-field-test-id"
            { ...rest }
          />
          { label && 
                <Label htmlFor={ name }>{ label }</Label>
            }